server/scripts/import-tests/index.js

```diff
@@ -159,277 +151,206 @@ const ariaat = {
-                );
+        const getReferenceUrl = referenceLine => {
+            let url = null;
+            if (referenceLine.length) {
+                const [referenceType, link] = referenceLine[0].split(',');
+                if (validUrl.isUri(link)) url = link;
+                else
+                    console.error(
+                        `WARNING: The ${referenceType} link ${link} is not valid for ${exampleName}. Not writing to database.`
+                    );
             }
-        }
+            return url;
+        };
+
+        const exampleUrl = getReferenceUrl(exampleRefLine);
+        const designPattern = getReferenceUrl(practiceGuidelinesRefLine);
+
+        let parsedTest = {
+            title: '',
+            gitRepo: ariaAtRepo,
+            directory: exampleDir,
+            minimumInputCount: 0,
+            maximumInputCount: 0,
+            testActions: [],
+            designPattern
+        };
+
+        // checking to see if unique testPlan row
+        const testPlanResult = await client.query(
+            'SELECT id, revision FROM "TestPlan" WHERE revision=$1 and "exampleUrl"=$2',
+            [commitHash, exampleUrl]
+        );
 
-        return example.id;
+        const testPlan = testPlanResult.rowCount
+            ? testPlanResult.rows[0]
+            : await db.TestPlan.create({
+                  title: exampleName,
+                  publishStatus: 'draft',
+                  revision: commitHash,
+                  sourceGitCommit: commitMsg,
+                  exampleUrl,
+                  createdAt: commitDate,
+                  parsedTest
+              });
+        return testPlan.id;
     },
 
     /**
-     * Gets the test id and adds the test record if it does not exist
-     * @param {string} testFullName - the name of the test
-     * @param {string} file - the relative path to the test in the repo
-     * @param {int} exampleID - foreign key into the apg_example table
-     * @param {int} testVersionID - foreign key into the test_version table
-     * @param {int} executionOrder - order of the tests (within APG pattern)
-     * @return {int} id
+     * Checks TestPlan.parsedTest.testActions to see if it has the relevant test actions to run the test and inserts it if not
+     * @param {string} testName - the name of the test
+     * @param {string} file - the relative path to the test file in the repository (ideally {@link https://github.com/w3c/aria-at.git})
+     * @param {number} testPlanId - TestPlan.id to be queried to update the TestPlan.parsedTest.testActions if necessary
+     * @param {string} commitHash - the hash of the latest version of tests pulled from the repository (ideally {@link https://github.com/w3c/aria-at.git})
+     * @param {string} commitMsg - the message of the latest version of tests pulled from the repository (ideally {@link https://github.com/w3c/aria-at.git})
```

Comment:
If you agree about the `sourceGitCommit` I think it would make sense to change this to `sourceGitCommitMessage`. 

About `msg`, one convention I always fight for is to spell out words!!! I think it reduces guesswork and makes variable names easier to remember. I fight for it, but that's definitely an optional suggestion!